src/transformers/generation/logits_process.py

```diff
@@ -1148,3 +1148,114 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         cond_logits, uncond_logits = scores.split(unguided_bsz, dim=0)
         scores = uncond_logits + (cond_logits - uncond_logits) * self.guidance_scale
         return scores
+
+
+class UnbatchedClassifierFreeGuidanceLogitsProcessor(LogitsProcessor):
+    r"""Logits processor for Classifier-Free Guidance (CFG). The processors
+    computes a weighted average across scores from prompt conditional and prompt unconditional (or negative) logits,
+    parameterized by the `guidance_scale`. The unconditional scores are computed internally by prompting `model` with
+    the `unconditional_ids` branch.
+
+    See [the paper](https://arxiv.org/abs/2306.17806) for more information.
+
+    Args:
+        guidance_scale (float):
+            The guidance scale for classifier free guidance (CFG). CFG is enabled by setting `guidance_scale > 1`.
+            Higher guidance scale encourages the model to generate samples that are more closely linked to the input
+            prompt, usually at the expense of poorer quality.
```

Comment:
>             prompt, usually at the expense of poorer quality.

You mentioned here that this can be overcome with a linear interpolation trick: https://github.com/huggingface/transformers/issues/24536#issuecomment-1611672874

Can we assume that quality will be maintained for higher guidance scales as a result?

Reply:
As @MPKonst remarked, the "linear trick" can be factored down into a simple guidance_strength reparameterization. So I removed it.